### PR TITLE
minor gas optimization + loop incremention consistency

### PR DESCRIPTION
**Pull request by wpoulin**
Improved loop incrementation consistency + minor gas optimization

### CODE REVIEW

**Comment by 0age** [comment]
Can you resolve the merge conflicts?

File: test/foundry/NonReentrant.t.sol [hunk 1]
```diff
@@ -56,8 +56,8 @@ contract NonReentrantTest is BaseOrderTest {
     }
 
     function testNonReentrant() public {
-        for (uint256 i; i < 7; i++) {
-            for (uint256 j; j < 10; j++) {
+        for (uint256 i; i < 7; ++i) {
```
Comment (axic) [inline test/foundry/NonReentrant.t.sol:59]: Which was more prevalent post or pre increment? This PR seems to change it at so many places I wonder if the new style was prevalent 😅

Comment (axic) [reply test/foundry/NonReentrant.t.sol:59]: There is also a claim about gas optimization, I only see the `i++` -> `++i` changes, is that really making any changes for gas?

Comment (wpoulin) [reply test/foundry/NonReentrant.t.sol:59]: I did update a few `var++ `to `++var` yesterday night, but this morning a bunch of PRs merged main and the gas optimization changes I made were already on master.

**Pull request merged**